src/main/java/org/springframework/data/jpa/provider/PersistenceProvider.java

```diff
@@ -355,6 +368,69 @@ public String getCountQueryPlaceholder() {
 	}
 
 	/**
+	 * Creates a dynamic {@link EntityGraph} from the given {@link JpaEntityGraph} information. 
+	 * 
+	 * @param em
+	 * @param jpaEntityGraph
+	 * @param entityType
+	 * @return
+	 * 
+	 * @since 1.9
+	 */
+	public EntityGraph<?> createDynamicEntityGraph(EntityManager em, JpaEntityGraph jpaEntityGraph, Class<?> entityType) {
+
+		Assert.isTrue(jpaEntityGraph.isDynamicEntityGraph(), "The given " + jpaEntityGraph + " is not dynamic!");
+
+		EntityGraph<?> entityGraph = em.createEntityGraph(entityType);
+
+		configureFetchGraphFrom(jpaEntityGraph, entityGraph);
+
+		return entityGraph;
+	}
+
+	
+	/**
+	 * Configures the given {@link EntityGraph} with the fetch graph information stored in {@link JpaEntityGraph}.
+	 * 
+	 * @param jpaEntityGraph
+	 * @param entityGraph
+	 */
+	/* visible for testing */
+	void configureFetchGraphFrom(JpaEntityGraph jpaEntityGraph, EntityGraph<?> entityGraph) {
+
+		String[] attributePaths = jpaEntityGraph.getAttributePaths().clone();
+
+		// sort to ensure that the intermediate entity subgraphs are created accordingly.
+		Arrays.sort(attributePaths);
+
+		// we build the entity graph based on the paths with highest depth first
+		for (int i = attributePaths.length - 1; i >= 0; i--) {
```

Comment:
Does it make sense to use `PropertyPath` or `PersistentPropertyPath` here to get some kind of verification whether the path segments actually refer to existing properties?